src/test/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScannerTest.java

```diff
@@ -0,0 +1,323 @@
+     * 3 - Echostep
+     * 4 - ParallelStep (stepstartnode)
+     * 6 - ParallelStep (StepStartNode) (start branch 1), ParallelLabelAction with branchname=1
+     * 7 - ParallelStep (StepStartNode) (start branch 2), ParallelLabelAction with branchname=2
+     * 8 - EchoStep (branch #1) - parentId=6
+     * 9 - StepEndNode (end branch #1) - startId=6
+     * 10 - EchoStep - parentId=7
+     * 11 - EchoStep
+     * 12 - ParallelStep (StepStartNode) - start inner parallel
+     * 14 - ParallelStep (StepStartNode) (start branch 2-1), parentId=12, ParallelLabellAction with branchName=2-1
+     * 15 - ParallelStep (StepStartNode) (start branch 2-2), parentId=12, ParallelLabelAction with branchName=2-2
+     * 16 - Echo (Branch2-1), parentId=14
+     * 17 - StepEndNode (end branch 2-1), parentId=16, startId=14
+     * 18 - SleepStep (branch 2-2) parentId=15
+     * 19 - EchoStep (branch 2-2)
+     * 20 - StepEndNode (end branch 2-2), startId=15
+     * 21 - StepEndNode (end inner parallel ), parentIds=17,20, startId=12
+     * 22 - StepEndNode (end parallel #2), parent=21, startId=7
+     * 23 - StepEndNode (end outer parallel), parentIds=9,22, startId=4
+     * 24 - Echo
+     * 25 - FlowEndNode
+     */
+    WorkflowRun NESTED_PARALLEL_RUN;
+
+    @Before
+    public void setUp() throws Exception {
+        r.jenkins.getInjector().injectMembers(this);
+
+        WorkflowJob job = r.jenkins.createProject(WorkflowJob.class, "SimpleParallel");
+        job.setDefinition(new CpsFlowDefinition(
+                "echo 'first'\n" +
+                        "def steps = [:]\n" +
+                        "steps['1'] = {\n" +
+                        "    echo 'do 1 stuff'\n" +
+                        "}\n" +
+                        "steps['2'] = {\n" +
+                        "    echo '2a'\n" +
+                        "    echo '2b'\n" +
+                        "}\n" +
+                        "parallel steps\n" +
+                        "echo 'final'"
+        ));
+        WorkflowRun b = r.assertBuildStatusSuccess(job.scheduleBuild2(0));
+        this.SIMPLE_PARALLEL_RUN = b;
+
+        job = r.jenkins.createProject(WorkflowJob.class, "NestedParallel");
+        job.setDefinition(new CpsFlowDefinition(
+                "echo 'first'\n" +
+                        "def steps = [:]\n" +
+                        "steps['1'] = {\n" +
+                        "    echo 'do 1 stuff'\n" +
+                        "}\n" +
+                        "steps['2'] = {\n" +
+                        "    echo '2a'\n" +
+                        "    def nested = [:]\n" +
+                        "    nested['2-1'] = {\n" +
+                        "        echo 'do 2-1'\n" +
+                        "    } \n" +
+                        "    nested['2-2'] = {\n" +
+                        "        sleep 1\n" +
+                        "        echo '2 section 2'\n" +
+                        "    }\n" +
+                        "    echo '2b'\n" +
```

Comment:
:ant: potentially misleading, since `echo '2b'` immediately follows `echo '2a'` in the graph. Would be more apparent if you were using the normal idiom:

```groovy
echo 'first'
parallel '1': {
  echo 'do 1 stuff'
}, '2': {
  echo '2a'
  echo '2b' // why?
  parallel '2-1': {
    echo 'do 2-1'
  }, '2-2': {
    sleep 1
    echo '2 section 2'
  }
}
echo 'final'
```